grid-client/calculator/calculate.go

```diff
@@ -143,6 +152,308 @@ func calculateCU(cru, mru int64) float64 {
 	return cu
 }
 
+// Calculates the cost of a public IP per month in USD.
+func (c *Calculator) calculateIPV4CostPerMonth() (float64, error) {
+	// cost in unit-USD per month
+	pricingPolicy, err := c.substrateConn.GetPricingPolicy(defaultPricingPolicyID)
+	if err != nil {
+		return 0, err
+	}
+	// cost in unit-USD per month
+	monthlyCost := float64(pricingPolicy.IPU.Value) * 24 * 30
+
+	return unitToUSD(monthlyCost), nil
+}
+
+// Calculates the overdue amount in TFT.
+//
+// The overdue amount basically is the sum of three parts:
+//
+//	1- Total over draft: is the sum of additional overdraft and standard overdraft.
+//	2- Unbilled NU: is the unbilled amount of network usage.
+//	3- The estimated cost of the contract for the total period: this part is dependant on the contract type and if the contract is on rented node or not.
+//
+// If the contract is rent contract, will add both of ipv4 cost and the total overdue of all associated contracts.
+// The total period is the time since the last billing added to Allowance period.
+// The resulting overdue amount represents the amount that needs to be addressed.
+func (c Calculator) CalculateContractOverdue(id uint64, allowance time.Duration) (int64, error) {
+	contract, err := c.substrateConn.GetContract(id)
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to get contract ID %d", id)
+	}
+
+	if contract.IsDeleted() {
+		return 0, ErrContractDeleted
+	}
+	contractInfo := contract.Contract
+
+	contractPaymentState, err := c.substrateConn.GetContractPaymentState(id)
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to get payment state for contract ID %d", id)
+	}
+
+	periodCostTFT, err := c.calculatePeriodCostTFT(time.Unix(int64(contractPaymentState.LastUpdatedSeconds), 0), contractInfo, allowance)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to calculate period cost")
+	}
+	// totalOverDraft represents the sum of standard and additional overdraft amounts for the contract in unit TFT
+	totalOverDraft := types.U128{Int: big.NewInt(0)}
+
+	var standardOverdraft types.U128
+	standardOverdraft.Int = big.NewInt(0)
+	if contractPaymentState.StandardOverdraft.Int != nil {
+		standardOverdraft.Int = contractPaymentState.StandardOverdraft.Int
+	}
```

Comment:
why not use this instead of introducing new variables 
```suggestion
	if contractPaymentState.standardOverdraft.Int == nil {
		contractPaymentState.Int = big.NewInt(0)
	}
```